Job "Build resctl-demo image" failing in the "Build ospack" step (time crate fails to build)

resctl-demo v2.2.5 (crates.io) uses time v0.3.34, which fails to build on rust >= 1.8.0 (https://github.com/rust-lang/rust/issues/127343). A simple workaround would be to pin the rust toolchain version to 1.79.0 until a new resctl-demo version is released and picks a more recent version of the time crate.

Failed job:
https://github.com/iocost-benchmark/resctl-demo-image-recipe/actions/runs/10445871057/job/28922450247

```
2024/08/19 02:09:48 install-resctl-demo |    Compiling time v0.3.34
2024/08/19 02:09:48 install-resctl-demo | error[E0282]: type annotations needed for `Box<_>`
2024/08/19 02:09:48 install-resctl-demo |   --> /home/demo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.34/src/format_description/parse/mod.rs:83:9
2024/08/19 02:09:48 install-resctl-demo |    |
2024/08/19 02:09:48 install-resctl-demo | 83 |     let items = format_items
2024/08/19 02:09:48 install-resctl-demo |    |         ^^^^^
2024/08/19 02:09:48 install-resctl-demo | ...
2024/08/19 02:09:48 install-resctl-demo | 86 |     Ok(items.into())
2024/08/19 02:09:48 install-resctl-demo |    |              ---- type must be known at this point
2024/08/19 02:09:48 install-resctl-demo |    |
2024/08/19 02:09:48 install-resctl-demo | help: consider giving `items` an explicit type, where the placeholders `_` are specified
2024/08/19 02:09:48 install-resctl-demo |    |
2024/08/19 02:09:48 install-resctl-demo | 83 |     let items: Box<_> = format_items
2024/08/19 02:09:48 install-resctl-demo |    |              ++++++++
2024/08/19 02:09:48 install-resctl-demo | 
```

--- a/overlays/resctl-demo/home/demo/.bin/install-resctl-demo
+++ b/overlays/resctl-demo/home/demo/.bin/install-resctl-demo
@@ -11,6 +11,10 @@ curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- $RUSTUP_ARG
 source $HOME/.cargo/env
 echo "source \$HOME/.cargo/env" >> $HOME/.bashrc
 
+# Fixup for resctl-demo v2.2.5: uses time v0.3.34, which fails to build
+# on rust >= 1.8.0
+rustup default 1.79.0
+
 # Append arguments to cargo install
 CARGO_ARGS=""
 if [ "$RESCTL_DEMO_SRC" = "git HEAD" ] ; then

